### PR TITLE
Cleanup the use of Double primitive wrapper

### PR DESCRIPTION
**Pull request by BradWalker**
As of Java 9, the VM can do a better job of handling the conversion from a Double primitive to a Double Object or vice versa.

This is an extension of the work done in #2498.

No tests have been touched as those expressly need to manage the conversion.

A couple of places I also cleaned up the Float.

---
**^Add meaningful description above**

By opening a pull request you confirm that, unless explicitly stated otherwise, the changes -

 - are all your own work, and you have the right to contribute them.
 - are contributed solely under the terms and conditions of the Apache License 2.0 (see section 5 of the license for more information).

Please make sure (eg. `git log`) that all commits have a valid name and email address for you in the Author field.

If you're a first time contributor, see the Contributing guidelines for more information.

### CODE REVIEW

**Review by lkishalmi — APPROVED** [review]
Looks fine, though I'd like the zero values to be consistent.

File: ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java [hunk 1]
```diff
@@ -176,7 +176,7 @@ private static double _renderPlainString(
                 }
 
                 double chWidth = wid / chars.length;
-                int estCharsToPaint = new Double(w / chWidth).intValue();
+                int estCharsToPaint = Double.valueOf(w / chWidth).intValue();
```
Comment (mbien) [inline ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java:179]: `(int) (w / chWidth)` would get rid of the temp boxing entirely

File: ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java [hunk 2]
```diff
@@ -841,7 +841,7 @@ If NO (it's a closing tag)
                             //Word wrap mode
                             goToNextRow = true;
 
-                            int lastChar = new Double(nextTag - estCharsOver).intValue();
+                            int lastChar = Double.valueOf(nextTag - estCharsOver).intValue();
```
Comment (mbien) [inline ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java:844]: same here

File: ide/db/src/org/netbeans/modules/db/explorer/dlg/CreateTableDialog.java [hunk 3]
```diff
@@ -471,7 +471,7 @@ public DataTable(TableModel model) {
                 Map cmap = ColumnItem.getColumnProperty(i);
                 col.setIdentifier(cmap.get("name")); //NOI18N
                 columnName = NbBundle.getMessage (CreateTableDialog.class, "CreateTable_" + i); //NOI18N
-                columnWidth = (new Double(getFontMetrics(getFont()).getStringBounds(columnName, getGraphics()).getWidth())).intValue() + 20;
+                columnWidth = Double.valueOf(getFontMetrics(getFont()).getStringBounds(columnName, getGraphics()).getWidth()).intValue() + 20;
```
Comment (mbien) [inline ide/db/src/org/netbeans/modules/db/explorer/dlg/CreateTableDialog.java:474]: i think `Double.valueOf` can be completely removed here, everything is an int already if i see this correctly.

File: enterprise/javaee.project/src/org/netbeans/modules/javaee/project/api/ant/ui/wizard/ImportBuildfile.java [hunk 4]
```diff
@@ -52,7 +52,7 @@ public ImportBuildfile(File buildFile, JButton okButton) {
     }
     
     private void resize() {
-        int width = (new Double(jLabelDesc.getFontMetrics(jLabelDesc.getFont()).getStringBounds(jLabelDesc.getText(), getGraphics()).getWidth() / 2.7)).intValue() + 40;
+        int width = Double.valueOf(jLabelDesc.getFontMetrics(jLabelDesc.getFont()).getStringBounds(jLabelDesc.getText(), getGraphics()).getWidth() / 2.7).intValue() + 40;
```
Comment (mbien) [inline enterprise/javaee.project/src/org/netbeans/modules/javaee/project/api/ant/ui/wizard/ImportBuildfile.java:55]:
```java
int width = (int) (jLabelDesc.getFontMetrics(jLabelDesc.getFont()).getStringBounds(jLabelDesc.getText(), getGraphics()).getWidth() / 2.7d) + 40
```

would remove temp boxing

File: ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java [hunk 5]
```diff
@@ -892,7 +892,7 @@ If NO (it's a closing tag)
                                 }
                             } else if (brutalWrap) {
                                 //wrap without checking word boundaries
-                                length = (new Double((w - widthPainted) / chWidth)).intValue();
+                                length = Double.valueOf((w - widthPainted) / chWidth).intValue();
```
Comment (mbien) [inline ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java:895]: -> `(int) ((w - widthPainted) / chWidth)`

File: ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java [hunk 6]
```diff
@@ -911,7 +911,7 @@ If NO (it's a closing tag)
 
                     if (strikethrough || underline) {
                         LineMetrics lm = fm.getLineMetrics(chars, pos, length - 1, g);
-                        int lineWidth = new Double(x + r.getWidth()).intValue();
+                        int lineWidth = Double.valueOf(x + r.getWidth()).intValue();
```
Comment (mbien) [inline ide/editor.completion/src/org/netbeans/modules/editor/completion/PatchedHtmlRenderer.java:914]: simple int addition, no need for boxing or a double

File: java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java [hunk 7]
```diff
@@ -437,8 +437,8 @@ public void startAdding(LayoutComponent[] comps,
             LayoutTestUtils.writeLayoutComponentArray(testCode, "comps", "lc");				    //NOI18N
             LayoutTestUtils.writeRectangleArray(testCode, "bounds", bounds);				    //NOI18N
             LayoutTestUtils.writeString(testCode, "defaultContId", defaultContId);			    //NOI18N         
-            testCode.add("Point hotspot = new Point(" + new Double(hotspot.getX()).intValue() + "," +	    //NOI18N
-			    new Double(hotspot.getY()).intValue() + ");");				    //NOI18N
+            testCode.add("Point hotspot = new Point(" + Double.valueOf(hotspot.getX()).intValue() + "," +	    //NOI18N
+			    Double.valueOf(hotspot.getY()).intValue() + ");");				    //NOI18N
```
Comment (mbien) [inline java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java:441]: (int) cast would do the job here too

File: java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java [hunk 8]
```diff
@@ -465,8 +465,8 @@ public void startMoving(String[] compIds, Rectangle[] bounds, Point hotspot) {
             testCode.add("{"); //NOI18N
             LayoutTestUtils.writeStringArray(testCode, "compIds", compIds); //NOI18N
             LayoutTestUtils.writeRectangleArray(testCode, "bounds", bounds); //NOI18N
-            testCode.add("Point hotspot = new Point(" + new Double(hotspot.getX()).intValue() + "," +  //NOI18N
-		    new Double(hotspot.getY()).intValue() + ");"); //NOI18N
+            testCode.add("Point hotspot = new Point(" + Double.valueOf(hotspot.getX()).intValue() + "," +  //NOI18N
+		    Double.valueOf(hotspot.getY()).intValue() + ");"); //NOI18N
```
Comment (mbien) [inline java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java:469]: same here

File: java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java [hunk 9]
```diff
@@ -519,8 +519,8 @@ public boolean startResizing(String[] compIds,
...
```
Comment (mbien) [inline java/form/src/org/netbeans/modules/form/layoutdesign/LayoutDesigner.java:523]: an here

File: java/java.j2seproject/src/org/netbeans/modules/java/j2seproject/ui/wizards/ImportFoldersPanel.java [hunk 10]
```diff
@@ -86,7 +86,7 @@ private void updateDistFolder() {
     }
 
     private void resize() {
-        int width = (new Double(message.getFontMetrics(message.getFont()).getStringBounds(message.getText(), getGraphics()).getWidth() / 2.7)).intValue() + 40;
+        int width = Double.valueOf(message.getFontMetrics(message.getFont()).getStringBounds(message.getText(), getGraphics()).getWidth() / 2.7).intValue() + 40;
```
Comment (mbien) [inline java/java.j2seproject/src/org/netbeans/modules/java/j2seproject/ui/wizards/ImportFoldersPanel.java:89]: see other comment, of the exact copy of this method

File: platform/o.n.swing.tabcontrol/src/org/netbeans/swing/tabcontrol/plaf/BaseTabLayoutModel.java [hunk 11]
```diff
@@ -123,7 +123,7 @@ protected int textHeight(int index, JComponent component) {
             //No need to calculate for every string
             String testStr = "Zgj"; //NOI18N
             Font f = getFont();
-            textHeight = new Double(f.getStringBounds(testStr, 
+            textHeight = Double.valueOf(f.getStringBounds(testStr, 
             BasicScrollingTabDisplayerUI.getOffscreenGraphics(component).getFontRenderContext()).getWidth()).intValue() + 2;
```
Comment (mbien) [inline platform/o.n.swing.tabcontrol/src/org/netbeans/swing/tabcontrol/plaf/BaseTabLayoutModel.java:127]: simple int addition again

File: platform/openide.awt/src/org/openide/awt/HtmlRenderer.java [hunk 12]
```diff
@@ -352,7 +352,7 @@ private static double _renderPlainString(
                 }
 
                 double chWidth = wid / chars.length;
-                int estCharsToPaint = new Double(w / chWidth).intValue();
+                int estCharsToPaint = Double.valueOf(w / chWidth).intValue();
```
Comment (mbien) [inline platform/openide.awt/src/org/openide/awt/HtmlRenderer.java:355]: convert to int cast

File: platform/openide.awt/src/org/openide/awt/HtmlRenderer.java [hunk 13]
```diff
@@ -1057,7 +1057,7 @@ If NO (it's a closing tag)
                             //Word wrap mode
                             goToNextRow = true;
 
-                            int lastChar = new Double(nextTag - estCharsOver).intValue();
+                            int lastChar = Double.valueOf(nextTag - estCharsOver).intValue();
```
Comment (mbien) [inline platform/openide.awt/src/org/openide/awt/HtmlRenderer.java:1060]: convert to int cast

File: platform/openide.awt/src/org/openide/awt/HtmlRenderer.java [hunk 14]
```diff
@@ -1108,7 +1108,7 @@ If NO (it's a closing tag)
                                 }
                             } else if (brutalWrap) {
                                 //wrap without checking word boundaries
-                                length = (new Double((w - widthPainted) / chWidth)).intValue();
+                                length = Double.valueOf((w - widthPainted) / chWidth).intValue();
```
Comment (mbien) [inline platform/openide.awt/src/org/openide/awt/HtmlRenderer.java:1111]: convert to int cast

File: platform/openide.awt/src/org/openide/awt/HtmlRenderer.java [hunk 15]
```diff
@@ -1127,7 +1127,7 @@ If NO (it's a closing tag)
 
                     if (strikethrough || underline || link) {
                         LineMetrics lm = fm.getLineMetrics(chars, pos, length - 1, g);
-                        int lineWidth = new Double(x + r.getWidth()).intValue();
+                        int lineWidth = Double.valueOf(x + r.getWidth()).intValue();
```
Comment (mbien) [inline platform/openide.awt/src/org/openide/awt/HtmlRenderer.java:1130]: convert to int cast

**Review by mbien — COMMENTED** [review]
I believe most of this boxing can be simply removed, see individual comments.

I hope i made no mistakes since I only looked at it via github.

the original author used
```java
new Double(d).intValue()
```
in a lot of places, which can be substituted with the cleaner

```java
(int) d
```

**Comment by mbien** [comment]
@BradWalker say if you need help with this one

**Comment by neilcsmith-net** [comment]
Given we're close to freeze, let's punt this to NB16.

**Comment by neilcsmith-net** [comment]
Given we're now close to NB16 freeze and there's been no further movement on this, going to close.  Re-open if necessary.

**Pull request closed without merging**